### PR TITLE
fix(shwap/bitswap): tune bitswap params

### PR DESCRIPTION
**Pull request by Wondertan**
List of notable changes:
* Fixes server-side peer serving prioritization by lowering message size.
   * Previously, some peer could max out server, while others rarely processed.
* Fine-tuned don't have timeout configuration
* Increasing number of workers

---

🎱 mb certified

### CODE REVIEW

**Review by cristaloleg — APPROVED** [review]

**Review by cristaloleg — CHANGES_REQUESTED** [review]
unapprove

File: share/shwap/p2p/bitswap/bitswap.go [hunk 1]
```diff
@@ -103,13 +121,16 @@ func NewClient(
...
 		// Prevents Has calls to Blockstore for metric that counts duplicates
 		// Unnecessary for our use case, so we can save some disk lookups.
 		client.WithoutDuplicatedBlockStats(),
-		client.WithDontHaveTimeoutConfig(dontHaveTimeoutConfig),
-		client.WithDisabledMessageQueueRebroadcast(true),
+
+		// These two options have mixed up named. One should be another and vice versa.
+		// TODO(@Wondertan): Make an issue
+		client.ProviderSearchDelay(broadcastDelay),
+		client.RebroadcastDelay(delay.Fixed(provSearchDelay)),
```
Comment (renaynay) [inline share/shwap/p2p/bitswap/bitswap.go:133]: This is super confusing, why can't you just fix it in this PR?

File: share/shwap/p2p/bitswap/bitswap.go [hunk 2]
```diff
@@ -133,6 +133,7 @@ func NewServer(
 		server.WithTargetMessageSize(targetMessageSize),
 		server.MaxOutstandingBytesPerPeer(outstandingBytesPerPeer),
 		server.WithWantHaveReplaceSize(replaceHasWithBlockMaxSize),
+		server.TaskWorkerCount(32),
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go]: Actually thanks for noticing this. This should be a constant with a comment

[the remark this answers is not in the document]
Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go]: Ah, wait that's a comment on the old code

File: share/shwap/p2p/bitswap/bitswap.go (hunk 1 above)
Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go:133]: This has to be fixed in Bitswap itself. They literally mixed up value.

* `WithPendussyFilled` fills renussy 
* `WithRenussyFilled`  fills pendussy

File: share/shwap/p2p/bitswap/bitswap.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go]: The default is 8

File: share/shwap/p2p/bitswap/bitswap.go [hunk 3]
```diff
@@ -17,58 +17,89 @@ import (
...
-	// Bigger number will speed transfers up if reading data from disk is fast. In our case, the
-	// Bitswap's size lookup via [Blockstore] will already cause underlying cache to keep the data,
-	// so reading up data is fast, and we can aim to pack as much as we can.
-	targetMessageSize = 1 << 20 // 1MB
-	// outstandingBytesPerPeer limits number of bytes queued for work for a peer across multiple requests.
-	// We set it to be equal to targetMessageSize * N, so there can max N messages being prepared for
-	// a peer at once.
-	outstandingBytesPerPeer = targetMessageSize * 4
+	// 8192 is 512 sampling requests of size 16.
+	maxServerWantListsPerPeer = 8192
```
[the remark this answers is not in the document]
Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go]: From sync discussion ^

There are recent workers as well. We can decrease it, tho.

File: share/shwap/p2p/bitswap/bitswap.go (hunk 1 above)
Comment (renaynay) [reply share/shwap/p2p/bitswap/bitswap.go:134]: Oh god 🤦🏻

**Review by renaynay — APPROVED** [review]
utack 🤞🏻

File: share/shwap/p2p/bitswap/bitswap.go (hunk 1 above)
Comment (cristaloleg) [reply share/shwap/p2p/bitswap/bitswap.go:134]: image

Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go:134]: The shit I had to go through guys to even realize that...

Comment (Wondertan) [reply share/shwap/p2p/bitswap/bitswap.go:134]: Ah, thats an old meme of ours ;)

**Review by walldiss — APPROVED** [review]

**Review by cristaloleg — APPROVED** [review]

**Pull request merged**